_posts/2022-03-16-aws-wafv2-association-with-api-gateway.md

```diff
@@ -0,0 +1,104 @@
+
+### Things to be aware of
+
+I got the tip on these ahead of my implementation thanks to Natalie’s [article](https://mechanicalrock.github.io/2019/09/26/aws-waf-association-with-api-gateway.html).
+
+- YAML indentation - I’d recommend installing [cfn-lint](https://github.com/aws-cloudformation/cfn-lint), a huge help for formatting YAML files and catching bugs early.
+- Formatting your API Gateway ARN, you will need this to associate it to the WebACL. This should look like:
+
+```yml
+arn:aws:apigateway:{region}::/restapis/{rest_api_id}/stages/{stage_name}
+```
+
+### Solution
+
+To configure your WAF you'll need to provision a WebACL then associate it to your API. In this case I was configuring a WAF to block SQL injection, however, this format could be used for other security protocols as well. Hence, the resources required at a minimum are:
+
+- [AWS::WAFv2::WebACL](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-wafv2-webacl.html)
+- [AWS::WAFv2::WebACLAssociation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-wafv2-webaclassociation.html)
+
+```yml
+WebACL:
+  Type: "AWS::WAFv2::WebACL"
+  Properties:
+    Name: WebACLSQLi
+    Scope: REGIONAL
+    Description: Web ACL to block SQL injection
+    DefaultAction:
+      Allow: {}
+    VisibilityConfig:
+      SampledRequestsEnabled: true
+      CloudWatchMetricsEnabled: true
+      MetricName: MyMetricName
+    Rules:
+      - Name: SQLInject-RuleSet
+        Priority: 0
+        Statement:
+          ManagedRuleGroupStatement:
+            VendorName: AWS
+            Name: AWSManagedRulesSQLiRuleSet
+        OverrideAction:
+          None: {}
+        VisibilityConfig:
+          SampledRequestsEnabled: true
+          CloudWatchMetricsEnabled: true
+          MetricName: SQLInjection-ruleset-metric
+
+WebACLAssociation:
+  Type: "AWS::WAFv2::WebACLAssociation"
+  Properties:
+    WebACLArn: !GetAtt WebACL.Arn
+    ResourceArn: !Ref ApiARN
+```
+
+I have used Amazon's Managed Rules to do the hard work of SQL injection blocking. You can see the other rule sets available [here](https://docs.aws.amazon.com/waf/latest/developerguide/aws-managed-rule-groups-list.html)
+
+If your API was created using the Serverless Framework you will need to run `npm i serverless-associate-waf`.
+In your serverless.yml you will need to utilize this plugin:
+
+```
+plugins:
+    - serverless-associate-waf
+```
+
+You will then need to append the following snippet:
+
+```
+custom:
+  associateWaf:
+    name:  WebACLSQLi
+    version: V2
+```
+
+The name must be the name you specified in your WebACL.
+
+### Wrapping up
+
+And voila, that's it. Simple right? It took me countless hours to get this configuration right. Hopefully after reading this post you won't have to!
```

Comment:
you won't have to!

You won't have to what? Waste hours too?

Reply:
good call- corrected